flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java

```diff
@@ -125,4 +160,82 @@ public void validateColumnListParams(
         // this makes it possible to ignore them in the validator and fall back to regular row types
         // see also SqlFunction#deriveType
     }
+
+    protected void registerNamespace(
+            @Nullable SqlValidatorScope usingScope,
+            @Nullable String alias,
+            SqlValidatorNamespace ns,
+            boolean forceNullable) {
+
+        // apply snapshot to SqlValidatorNameSpace
+        // Time travel only supports constant expressions, so we need to investigate scenarios
+        // where the period of Snapshot is a SqlIdentifier.
+        if (usingScope != null
```

Comment:
Extract the check logic to a new method, and please add some comments for the new method.

Reply:
Add a new method getTimeTravelNode to getSnapshotNode instread of  check only.